File: Golang/strategy/fly_behavior.go
package strategy

import "fmt"

type FlyBehavior interface {
	Fly()
}

type FlyWithWings struct{}

func(fww FlyWithWings) Fly() {
	fmt.Println("I am flying!")
}

type FlyNoWay struct{}

func(fnw FlyNoWay) Fly() {
	fmt.Println("I can't fly")
}

type FlyRocketPowered struct{}

func(frp FlyRocketPowered) Fly() {
	fmt.Println("I'm flying with a rocket!")
}

File: Python/strategy/fly_behavior.py
from abc import ABC, abstractmethod


class FlyBehavior(ABC):

    @abstractmethod
    def fly(self) -> None:
        raise NotImplementedError


class FlyWithWings(FlyBehavior):
    def fly(self) -> None:
        print("I am flying with wings!")


class FlyNoWay(FlyBehavior):
    def fly(self) -> None:
        print("I can't fly")


class FlyRocketPowered(FlyBehavior):
    def fly(self) -> None:
        print("I'm flying with a rocket!")
